src/test/java/school/hei/haapi/integration/TeacherIT.java

```diff
@@ -277,6 +277,39 @@ void manager_write_update_ok() throws ApiException {
     assertEquals(toUpdate, updated.get(0));
   }
 
+  @Test
+  void manager_write_update_without_email_field_ko() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    UsersApi api = new UsersApi(manager1Client);
+
+    List<Teacher> toUpdate =
+        api.createOrUpdateTeachers(List.of(someCreatableTeacher()));
+
+    Teacher toUpdate0 = toUpdate.get(0);
+    toUpdate0.setEmail(null);
+
+    assertThrowsApiException(
+        "{\"type\":\"400 BAD_REQUEST\",\"message\":\"Email is mandatory\"}",
+        () -> api.createOrUpdateTeachers(toUpdate));
+  }
+
+  @Test
+  void manager_write_update_without_names_ko() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    UsersApi api = new UsersApi(manager1Client);
+
+    List<Teacher> toUpdate =
+        api.createOrUpdateTeachers(List.of(someCreatableTeacher()));
```

Comment:
Useless to make an update call. Just create an non-creatableTeacher from an creatableTeacher. Also, divide your tests in three parts: given, when and then. Here, you have 4 parts.